examples/reshare/src/orchestrator/actor.rs

```diff
@@ -0,0 +1,300 @@
+            metadata::Config {
+                partition: format!("{}-metadata", self.partition_prefix),
+                codec_config: (),
+            },
+        )
+        .await
+        .expect("failed to initialize orchestrator metadata");
+
+        // Enter the initial epoch
+        let initial_epoch = metadata.get(&METADATA_KEY).cloned().unwrap_or(0);
+        self.enter_epoch(
+            initial_epoch,
+            initial_poly,
+            initial_share,
+            Set::from_iter(initial_participants),
+            &mut metadata,
+            &mut pending_mux,
+            &mut recovered_mux,
+            &mut resolver_mux,
+        )
+        .await;
+
+        // Wait for instructions to transition epochs.
+        while let Some(transition) = self.mailbox.next().await {
+            if transition.epoch <= self.epoch {
+                continue;
+            }
+
+            // Enter the new epoch.
+            self.enter_epoch(
+                transition.epoch,
+                transition.poly,
+                transition.share,
+                transition.participants,
+                &mut metadata,
+                &mut pending_mux,
+                &mut recovered_mux,
+                &mut resolver_mux,
+            )
+            .await;
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    async fn enter_epoch(
+        &mut self,
+        epoch: Epoch,
+        polynomial: Public<V>,
+        share: Option<group::Share>,
+        participants: Set<C::PublicKey>,
+        metadata: &mut Metadata<ContextCell<E>, FixedBytes<1>, Epoch>,
+        pending_mux: &mut MuxHandle<
+            impl Sender<PublicKey = C::PublicKey>,
+            impl Receiver<PublicKey = C::PublicKey>,
+        >,
+        recovered_mux: &mut MuxHandle<
+            impl Sender<PublicKey = C::PublicKey>,
+            impl Receiver<PublicKey = C::PublicKey>,
+        >,
+        resolver_mux: &mut MuxHandle<
+            impl Sender<PublicKey = C::PublicKey>,
+            impl Receiver<PublicKey = C::PublicKey>,
+        >,
+    ) {
+        let _ = metadata.put_sync(METADATA_KEY, epoch).await;
+        self.epoch = epoch;
+
+        // Stop the previous consensus engine, if there is one.
+        if let Some(engine) = self.engine.take() {
+            engine.abort();
+        }
+
+        let supervisor = Supervisor::<V, C::PublicKey>::new(polynomial, participants.into(), share);
+        let engine = threshold_simplex::Engine::new(
+            self.context.with_label("consensus_engine"),
+            threshold_simplex::Config {
+                crypto: self.signer.clone(),
+                blocker: self.oracle.clone(),
+                automaton: self.application.clone(),
+                relay: self.application.clone(),
+                reporter: self.marshal.clone(),
+                supervisor,
+                partition: format!("consensus_{}_{}", self.signer.public_key(), epoch),
```

Comment:
nit: we should use `partition_prefix` to delineate between participants not their public key here.